modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/WorkflowOperationInstance.java

```diff
@@ -21,98 +21,394 @@
 
 package org.opencastproject.workflow.api;
 
+import java.util.Collections;
 import java.util.Date;
-import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
 
-import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+import javax.persistence.Access;
+import javax.persistence.AccessType;
+import javax.persistence.CollectionTable;
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.Lob;
+import javax.persistence.ManyToOne;
+import javax.persistence.MapKeyColumn;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
 
 /**
  * A workflow operation belonging to a workflow instance.
  */
-@XmlJavaTypeAdapter(WorkflowOperationInstanceImpl.Adapter.class)
-public interface WorkflowOperationInstance extends Configurable {
-  enum OperationState {
+@Entity(name = "WorkflowOperationInstance")
+@Access(AccessType.FIELD)
+@Table(name = "oc_workflow_operation")
+public class WorkflowOperationInstance implements Configurable {
+  public enum OperationState {
     INSTANTIATED, RUNNING, PAUSED, SUCCEEDED, FAILED, SKIPPED, RETRY
   }
 
+  @Id
+  @GeneratedValue
+  @Column(name = "id")
+  private Long id;
+
+  @Column(name = "template")
+  protected String template;
+
+  @Column(name = "job")
+  protected Long jobId;
+
+  @Column(name = "state")
+  protected OperationState state;
+
+  @Column(name = "description")
+  protected String description;
+
+  @ElementCollection
+  @CollectionTable(
+          name = "oc_workflow_operation_configuration",
+          joinColumns = @JoinColumn(name = "workflow_operation_id")
+  )
+  @MapKeyColumn(name = "key_part", nullable = false)
+  @Lob
+  @Column(name = "value_part")
+  protected Map<String, String> configurations;
+
+  @Column(name = "failOnError")
+  protected boolean failWorkflowOnException;
+
+  @Column(name = "if_condition")
+  protected String executeCondition;
```

Comment:
Mixing camel case (which usually is not carried to the databases) and snake case.
I suggest using the latter for everything.